eagles-food-backend/Services/UserRepository/UserService.cs

```diff
@@ -170,6 +170,58 @@ public UserService(LunchDbContext db_context, IMapper mapper, AuthenticationClas
             return response;
         }
 
+        // update a user
+        public async Task<Response<Dictionary<string, string>>> UpdateUserProfile(
+            int userId, UpdateUserDTO model)
+        {
+            Response<Dictionary<string, string>> response = new();
+            User? user = await db_context.Users.FindAsync(userId);
+
+            try
+            {
+                // ensure user exists
+                if (user is null)
+                {
+                    response.success = false;
+                    response.message = "User not found";
+                    response.statusCode = HttpStatusCode.NotFound;
+                    response.data = new Dictionary<string, string>() {
+                        { "id", userId.ToString() }
+                    };
+
+                    return response;
+                }
+
+                db_context.Entry(user).State = EntityState.Modified;
+                user.LastName = model.LastName ?? user.LastName;
+                user.FirstName = model.FirstName ?? user.FirstName;
+                user.Phone = model.Phone ?? user.Phone;
+                user.ProfilePic = model.ProfilePic ?? user.ProfilePic;
+
+
+                await db_context.SaveChangesAsync();
+
+                response.success = true;
+                response.data = new Dictionary<string, string>() {
+                    { "Email", user.Email },
+                    { "Last name", user.LastName },
+                    { "First name", user.FirstName },
+                    { "Phone", user.Phone }
+                };
```

Comment:
mobile devs want more stuff returned, check master to see now...we're defaulting to tranforming a User to a dictionary, then removing any sensitive fields

woulda done that with a nice DTO but all the good naes are taken